streams/src/main/java/org/apache/kafka/streams/processor/PartitionGrouper.java

```diff
@@ -26,19 +26,71 @@
 
 public abstract class PartitionGrouper {
 
-    protected Map<Integer, Set<String>> topicGroups;
+    public static class TopicsInfo {
+        public Set<String> sourceTopics;
+        public Set<String> stateTopics;
+
+        public TopicsInfo(Set<String> sourceTopics, Set<String> stateTopics) {
+            this.sourceTopics = sourceTopics;
+            this.stateTopics = stateTopics;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o instanceof TopicsInfo) {
+                TopicsInfo other = (TopicsInfo) o;
+                return other.sourceTopics.equals(this.sourceTopics) && other.stateTopics.equals(this.stateTopics);
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            long n = ((long) sourceTopics.hashCode() << 32) | (long) stateTopics.hashCode();
+            return (int) (n % 0xFFFFFFFFL);
+        }
+    }
+
+    public static class TasksInfo {
+        public Map<TaskId, Set<TopicPartition>> partitionsForTask;
+        public Map<String, Set<TaskId>> tasksForState;
```

Comment:
Yeah I agree, though the problem is that only TopologyBuilder knows the state topics to tasks mapping, and it is only accessible to PartitionGrouper of the StreamThread. 

I will change it to make the topology return only this map via StreamThread to KafkaStreamingPartitionAssignor.